src/main/java/org/neo4j/sql2cypher/SQLToCypher.java

```diff
@@ -44,38 +44,87 @@
 import org.neo4j.cypherdsl.core.renderer.Configuration;
 import org.neo4j.cypherdsl.core.renderer.Renderer;
 
-import static org.jooq.impl.DSL.createTable;
-
 /**
  * Quick proof of concept of a jOOQ/Cypher-DSL based SQL to Cypher translator
  * @author Lukas Eder
  * @author Michael J. Simons
  */
 public class SQLToCypher {
 
+
     public static void main(String[] args) {
-        System.out.println(new SQLToCypher().convert("""
+        System.out.println(new SQLToCypher().using(new Config()).convert("""
             SELECT t.a, t.b
             FROM my_table AS t
             WHERE t.a = 1
             """));
     }
 
-    // Meta data can be added via .meta(createTable("t").column("a", SQLDataType.INTEGER))
-    private final Parser parser = DSL.using(SQLDialect.DEFAULT).parser();
-    private final Renderer renderer = Renderer.getRenderer(Configuration.prettyPrinting());
-    private final Map<Table<?>, Node> tables = new HashMap<>();
+    private Config config = new Config();
+
+    static class Context {
+        private final Map<Table<?>, Node> tables = new HashMap<>();
+
+        void addTable(Table<?> table, Node node) {
+            this.tables.put(table,node);
+        }
+
+        private Node getNode(Table<?> t) {
+            return tables.get(t);
+        }
+    }
+
+    static class Config {
+        private final SQLDialect dialect;
+        private final Configuration cypherDslConfig;
+
+        public Config() {
+            this.dialect = SQLDialect.DEFAULT;
+            this.cypherDslConfig = Configuration.prettyPrinting();
+        }
+
+        public Config(SQLDialect dialect, Configuration cypherDslConfig) {
+            this.dialect = dialect;
+            this.cypherDslConfig = cypherDslConfig;
+        }
+
+        public SQLDialect getDialect() {
+            return dialect;
+        }
+
+        public Configuration getCypherDslConfig() {
+            return cypherDslConfig;
+        }
+    }
+    public SQLToCypher using(Config config) {
+        this.config = config;
+        return this;
+    }
 
     // Unsure how thread safe this should be (wrt the node lookup table), but this here will do for the purpose of adding some tests
     public String convert(String sql) {
-        Select<?> query = parser.parseSelect(sql);
-        return renderer.render(statement(query));
+        Context ctx = new Context();
+        Select<?> query = parse(sql);
+        ResultStatement statement = statement(query, ctx);
+        return render(statement);
     }
 
-    private ResultStatement statement(Select<?> x) {
+    Select<?> parse(String sql) {
+        Parser parser = DSL.using(config.getDialect()).parser();
```

Comment:
In the context of this utility, the jOOQ `SQLDialect` won't be too useful (except for debugging, perhaps?) as it only affects how jOOQ *generates* SQL. What you want to set here is the `Settings.parseDialect`:

```java
DSL.using(dialect, new Settings().withParseDialect(dialect))
```

Reply:
Ok, added it in the new version.